### PR TITLE
feat(Adds pending indicator state

### PR DESCRIPTION
**Pull request by tsipoy**
Hello everyone,

We have just finished pending indicator state. Please have a look at it and your comments will be appreciated.

Refs: #45e5qe,
Refs: #4deynv and pending indicator with categories list 

Thank you.

### CODE REVIEW

**Review by vakodrazan — APPROVED** [review]

**Review by bakozetra — APPROVED** [review]

File: src/Pages/HomePage.tsx [hunk 1]
```diff
@@ -23,9 +24,13 @@ import { TopCategories } from '../components/TopCategories/TopCategories'
 import LearnMoreBanner from '../components/LearnMoreBanner/LearnMoreBanner'
 import { history } from '../redux/store'
 export const HomePage = () => {
+
   const dispatch = useAppDispatch()
   const offers = useSelector(adsSelector)
-  const categories = useSelector(categoriesSelector)  
+  const categories = useSelector(categoriesSelector) 
+  const categoriesStatus = useSelector(categoriesStatusSelector)
+  const pendingStatus = useSelector(adsStatusSelector)
```
Comment (andrzej-onja) [inline src/Pages/HomePage.tsx]: to follow one convention this should be `offersStatus`

File: src/components/CategoryList/CategoryList.tsx [hunk 2]
```diff
@@ -19,9 +21,11 @@ export const CategoryList: React.FC<CategoryListProps> = ({
   primary,
   categories,
   selectCategory,
+  pendingStatus
 }) => {
   return (
     <CategoryListContainer>
+      {pendingStatus === 'loading' && <PendingIndicator alt={"Pending indicator"} size={64} />}
```
Comment (andrzej-onja) [inline src/components/CategoryList/CategoryList.tsx]: as size use enum IconSize

File: src/Pages/OnboardingRegistration.tsx [hunk 3]
```diff
@@ -113,7 +111,12 @@ export const OnboardingRegistration: FC<OnboardingProp> = ({ role }) => {
       onClickRegister={() => null}
       onSubmit={handleRegister}
       errorMessage={errorMessages}
-      loadingButtonIcon={userStatus === 'loading' ? Loading : undefined}
+      loadingButtonIcon={userStatus === "loading"
+        ? <PendingIndicator
+            alt={"Pending indicator"}
+            size={20}
```
Comment (andrzej-onja) [inline src/Pages/OnboardingRegistration.tsx]: as size use enum IconSize

File: src/components/ListOffers/ListOffers.tsx [hunk 4]
```diff
@@ -20,11 +21,13 @@ interface OfferData {
...
 export interface ListOffersProps {
   offers: OfferData[];
+  pendingStatus: string;
 }
-const ListOffers: React.FC<ListOffersProps> = ({ offers }) => {
+const ListOffers: React.FC<ListOffersProps> = ({ offers,pendingStatus }) => {
   
   return (
     <>
+      {pendingStatus === "loading" && <PendingIndicator alt={"Pending indicator"} size={164} />}
```
Comment (andrzej-onja) [inline src/components/ListOffers/ListOffers.tsx:30]: as size use enum IconSize

File: src/components/PendingIndicator/PendingIndicator.tsx [hunk 5]
```diff
@@ -14,20 +14,16 @@ export const IconSize = {
...
-export const PendingIndicator: React.FC<PendingProps> = ({src, alt, size}) => {    
+export const PendingIndicator: React.FC<PendingProps> = ({ alt, size}) => {    
     return (
-        <Button>
+        <PendingImgWrapper>
             <PendingImg 
                 style={{maxWidth: `${size}px`}}
-                src={src} 
+                src={vamalakaLoading} 
                 alt={alt}
```
Comment (andrzej-onja) [inline src/components/PendingIndicator/PendingIndicator.tsx:17]: so far alt is always the same so create default value for it ({ alt, size='Pending indicator'}). This way you dont have to repeat it on every usage

File: src/components/CategoryList/CategoryList.tsx (hunk 2 above)
Comment (andrzej-onja) [reply src/components/CategoryList/CategoryList.tsx]: We are doing lot of checks against the current status and its also used in slices.
Please create folder with file constants/index.ts
export enum FETCH_STATUS {
LOADING: 'loading'
IDLE:'idle' 
}

Replace all hardcoded strings and use enum (slices, components)

File: src/components/CategoryList/CategoryList.tsx [hunk 6]
```diff
@@ -19,9 +22,11 @@ export const CategoryList: React.FC<CategoryListProps> = ({
   primary,
   categories,
   selectCategory,
+  status: categoriesStatus
 }) => {
   return (
     <CategoryListContainer>
+      {categoriesStatus === 'loading' && <PendingIndicator alt={"Pending indicator"} size={ICON_SIZE.MEDIUM_ICON} />}
```
Comment (andrzej-onja) [inline src/components/CategoryList/CategoryList.tsx]: `categoriesStatus === 'loading'`

should be replaced with

`categoriesStatus === FETCH_STATUS.LOADING` 

All instances of string `"loading"` should be replaced with FETCH_STATUS.LOADING

**Review by andrzej-onja — APPROVED** [review]

**Pull request merged**